test/transactions_test.rb

```diff
@@ -196,6 +196,16 @@ def test_multi_with_a_block_yielding_the_client
     assert_equal "s1", r.get("foo")
   end
 
+  def test_multi_with_interrupt_preserves_client
+    original = r._client
+    begin
+      Redis::Pipeline::Multi.stubs(:new).raises(Interrupt)
+      r.multi {}
+    rescue Interrupt
```

Comment:
It would be better to use `assert_raises Interrupt` as to ensure that your stub is indeed called.

Reply:
Roger, changed.